src/main/java/dk/kb/present/copyright/CopyrightAccessExtractor.java

```diff
@@ -0,0 +1,364 @@
+            Element accessConditionElement = (Element) accessCondititions.item(i);
+           
+            Node typeNode = accessConditionElement.getAttributes().getNamedItem("type");
+            if (typeNode != null) {
+                accessCondition.setType(typeNode.getNodeValue());     
+            }
+            
+            Node displayNode = accessConditionElement.getAttributes().getNamedItem("displayLabel");
+            if (displayNode != null) {
+                accessCondition.setDisplayLabel(displayNode.getNodeValue());                
+            }
+
+            NodeList copyrightList = accessConditionElement.getElementsByTagName("cdl:copyright");  
+           
+            
+            if (copyrightList.getLength()==0) { //Text!               
+                String accessContentText= accessConditionElement.getTextContent();
+                if (accessContentText != null) {
+                    accessContentText=accessContentText.trim();
+                }
+                accessCondition.setValue(accessContentText);                
+            }            
+            if (copyrightList.getLength() ==1) { //XML      
+                String publication_status = copyrightList.item(0).getAttributes().getNamedItem("publication.status").getNodeValue();
+                String copyright_status = copyrightList.item(0).getAttributes().getNamedItem("copyright.status").getNodeValue();
+                accessCondition.setCopyrightPublicationStatus(publication_status);
+                accessCondition.setCopyrightStatus(copyright_status);
+            }
+            else if (copyrightList.getLength() >1) { //TODO deleted later. Just sanity check for new
+                throw new InvalidArgumentServiceException("UPS... Need to handle multiple elements for 'cdl:copyright'");
+            }
+
+
+            //Can be empty list
+            ArrayList<CreatorPerson> creatorPersons = buildPersons(accessConditionElement);
+            accessCondition.setCreatorPersonList(creatorPersons);
+
+           //Can be empty list
+            ArrayList<CreatorCorporate> creatorCorporate = buildCooperate(accessConditionElement);
+            
+            accessCondition.setCreatorCorporateList(creatorCorporate);
+            
+            accessConditionList.add(accessCondition);                     
+        }
+
+        return accessConditionList ;
+    }
+
+    /*
+   <cdl:creator.person>
+     <cdl:name>Clemens, Johann Friderich</cdl:name>
+     <cdl:year.birth>1748-11-29</cdl:year.birth>
+     <cdl:year.death>1831-11-5</cdl:year.death>
+   </cdl:creator.person>        
+     */     
+    public static ArrayList<CreatorPerson> buildPersons( Element accessConditionElement){       
+        ArrayList<CreatorPerson> personList = new  ArrayList<CreatorPerson>();
+
+        NodeList creatorPerson = accessConditionElement.getElementsByTagName("cdl:creator.person");
+        if (creatorPerson != null) {
```

Comment:
General trick: If you have the structure
```
List<Whatever> myList = new ArrayList<Whataver>();
if (foo != null) {
  ...lotsa lines...
}
return myList;
```
then rewriting to 
```
if (foo == null) {
  return Collections.emptyList();
}

List<Whatever> myList = new ArrayList<Whataver>();
...lotsa lines...
return myList;
```
saves an indent, makes it very clear what happens if `foo` is missing and makes the code flow easier to read. Same principle in `buildCooperate`.

Reply:
agree. fixed